src/validators/ResponseValidator.js

```diff
@@ -0,0 +1,37 @@
+
+const Validator = require('./Validator');
+
+class ResponseValidator extends Validator {
+    constructor(schema) {
+        super(responseValidator, schema);
+    }
+}
+function responseValidator(response, data) {
```

Comment:
I don't understand much what is going on here please explain to me further when you can.
It seems that you build a new type of Validator which extends Validator but can also be either oneOf or SimpleValidator so how many classes are involved?